addon/components/bs-accordion/item/title.js

```diff
@@ -32,13 +30,12 @@ export default class AccordionItemTitle extends Component {
    * @event onClick
    * @public
    */
-  onClick() {}
 
   @action
   handleClick(e) {
     e.preventDefault();
-    if (!this.disabled) {
-      this.onClick();
+    if (!this.args.disabled) {
+      this.args.onClick?.();
     }
   }
```

Comment:
Wondering how hard we should try to push for template-only components. Template-only components should provide a better performance. It should be possible in this case if introducing a `bs-noop` template helper and using `{{action}}` helper. It would look like this:

```hbs
{{on "click" (if @disabled (bs-noop) (action @onClick preventDefault=true))}}
```

Reply:
> Template-only components should provide a better performance

I don't think that will have any meaningful effect here, unless you are rendering hundreds or thousands of accordion items!? :) Also you would have to account for the additional cycles that the template helpers need, which also occur even when the item is never clicked.

Reply:
Agree.

We may want to revisit as soon as [Contextual Helper and Modifier]() are available in all supported versions. If I got https://github.com/emberjs/rfcs/issues/660#issuecomment-689013203 right, it will make it both more readable and performant:

```hbs
{{unless @disabled (on "click" (action @onClick preventDefault=true))}}
```